### PR TITLE
feat(esl-related-target): make related target able to targeting multiple related togglable

### PR DESCRIPTION
**Pull request by Natalie-Smirnova**
- was made related target able to target multiple related togglable
- was introduced `esl-related-target-action` with  `'hide'|'show'|'all'(default)` as a values

Closes: #427

### CODE REVIEW

**Review by ala-n — APPROVED** [review]

File: build/jest.html-loader.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  process(sourceText) {
+    return {
+      code: `module.exports = ${JSON.stringify(sourceText)};`,
```
Comment (yadamskaya) [inline build/jest.html-loader.js:4]: minor
```suggestion
      code: `module.exports = ${JSON.stringify(sourceText)};`
```

**Review by yadamskaya — APPROVED** [review]

File: build/jest.html-loader.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    };
+  },
```
Comment (dshovchko) [inline build/jest.html-loader.js:6]:
```suggestion
  }
```

File: src/modules/esl-related-target/core/esl-related-target.ts [hunk 3]
```diff
@@ -39,14 +45,16 @@ export class ESLRelatedTarget extends ESLMixinElement {
   /** Processes {@link ESLToggleable} show event */
   @listen('esl:show')
   protected onShow(e: CustomEvent): void {
+    if (this.action === 'hide') return;
     const {params} = e.detail;
-    this.$relatedTarget?.show({params: this.mergeLocalParams(params)});
+    this.$targets.forEach(target => target.show({params: this.mergeParams(params)}));
```
Comment (dshovchko) [inline src/modules/esl-related-target/core/esl-related-target.ts:50]:
```suggestion
    this.$targets.forEach((target) => target.show({params: this.mergeParams(params)}));
```

File: src/modules/esl-related-target/core/esl-related-target.ts [hunk 4]
```diff
@@ -39,14 +45,16 @@ export class ESLRelatedTarget extends ESLMixinElement {
...
   }
 
   /** Processes {@link ESLToggleable} hide event */
   @listen('esl:hide')
   protected onHide(e: CustomEvent): void {
+    if (this.action === 'show') return;
     const {params} = e.detail;
-    this.$relatedTarget?.hide({params: this.mergeLocalParams(params)});
+    this.$targets.forEach(target => target.hide({params: this.mergeParams(params)}));
```
Comment (dshovchko) [inline src/modules/esl-related-target/core/esl-related-target.ts:58]:
```suggestion
    this.$targets.forEach((target) => target.hide({params: this.mergeParams(params)}));
```

**Review by dshovchko — APPROVED** [review]

File: pages/views/examples/accordion.njk [hunk 5]
```diff
@@ -184,3 +184,117 @@ aside:
     </esl-panel-group>
   </div>
 </section>
+<section class="row">
+  <div class="col-12">
+    <h2>Simple multiple accordion (panels 1 and 1.3 contain esl-related-target attribute)</h2>
```
Comment (NastaLeo) [inline pages/views/examples/accordion.njk:189]:
```suggestion
    <h2>Simple multiple accordion (panel 1 contains esl-related-target attribute)</h2>
```

File: src/modules/esl-related-target/README.md [hunk 6]
```diff
@@ -1,12 +1,26 @@
...
+The attribute's value defines a selector to find the corresponding element.
+Activation (set open state) or deactivation of the related target happens when the corresponding `esl:show` or 
+`esl:hide` DOM events are dispatched on the current host element.
+
+Mixin supports `esl-related-target-action` attribute to define required for synchronization actions:
+ - `all` (default) - to synchronize both: show and hide events of observed toggleable.
+ - `hide` - to synchronize only hide events of the observed toggleable
+ - `show` - to synchronize only show events of the observed toggleable
+
+To make cross-reference (bidirectional binding) of toggleables you need to cross-reference both of the tooglables:
```
Comment (NastaLeo) [inline src/modules/esl-related-target/README.md:22]:
```suggestion
To make cross-reference (bidirectional binding) of toggleables you need to cross-reference both of the toogleables:
```

**Review by NastaLeo — APPROVED** [review]

**Review by ala-n — CHANGES_REQUESTED** [review]
Example updated to the real use-case
PR rejected as esl-related-target doesn't work on the real case (see added nested accordion example)

File: src/modules/esl-related-target/core/esl-related-target.ts [hunk 7]
```diff
@@ -19,17 +21,21 @@ export class ESLRelatedTarget extends ESLMixinElement {
...
-  /** @returns related toggleable instance */
-  public get $relatedTarget(): ESLToggleable | null {
+  /** @returns related toggleable instances */
+  public get $targets(): ESLToggleable[] {
     const {selector} = this;
-    if (!selector) return null;
-    const target = ESLTraversingQuery.first(selector);
-    return target instanceof ESLToggleable ? target : null;
+    if (!selector) return [];
+    const targets = ESLTraversingQuery.all(selector);
```
Comment (ala-n) [inline src/modules/esl-related-target/core/esl-related-target.ts:31]: Incorrect usage of Traversing Query base element is missing to support relative pathes

**Review by ala-n — APPROVED** [review]

File: pages/views/examples/accordion.njk [hunk 8]
```diff
@@ -184,3 +192,123 @@ aside:
...
 </section>
+
+<section class="row">
+  <div class="col-12">
+    <h2>Nested accordion with auto-closable nested panels (esl-related-target)</h2>
+    {#    {% code 'html', 'code-block' %} #}
+    {#      <esl-panel-group class="accordion-group mb-4" min-open-items="0" max-open-items="all"> #}
+    {#        #}
+    {#      </esl-panel-group> #}
+    {#    {% endcode %} #}
```
Comment (abarmina) [inline pages/views/examples/accordion.njk:203]: do we need these comments?

**Review by abarmina — APPROVED** [review]

**Pull request merged**